### PR TITLE
Feature/exercise

### PR DESCRIPTION
**Pull request by futianshen**

### CODE REVIEW

File: src/components/exercise/ExerciseResultBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+}> = ({ exercises, passingScore, nextProgramContentId, onSetStatusAnswering, onSetStatusReview }) => {
+  const { formatMessage } = useIntl()
+  const history = useHistory()
+  const {
+    params: { programContentId: currentContentId },
+    url,
+  }: {
+    params: { programContentId: string }
+    url: string
+  } = useRouteMatch()
```
Comment (Elantris) [inline src/components/exercise/ExerciseResultBlock.tsx:74]: NITS: add type parameter to useRouteMatch like

```
useRouteMatch<{ programContentId: string }>()
```

File: src/components/exercise/ExerciseQuestionBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    options: {
+      answer: string
+      isAnswer: boolean
+      isSelected: boolean
+    }[]
+    question: string
+    detail: string
+    score: number
+  }[]
+  onSetStatusResult: () => void
```
Comment (Elantris) [inline src/components/exercise/ExerciseQuestionBlock.tsx:57]: NITS: it is better to use semantic word like `onSeeResult()`, `onFinish()` or `onCheckScore()`

File: src/components/exercise/ExerciseQuestionBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        options: {
+          answer: string
+          isAnswer: boolean
+          isSelected: boolean
+        }[]
+        detail: string
+        score: number
+      }[]
+    >
+  >
```
Comment (Elantris) [inline src/components/exercise/ExerciseQuestionBlock.tsx:72]: the event should be simplified and focus on action meaning: on which question what options are chosen

```
onSetAnswer: (question, options) => void
// index of question or any key to indicate the question
```

**Comment by futianshen** [comment]
[【測驗】前台切版-測驗](https://trello.com/c/TruAH96c/1799-%E3%80%90%E6%B8%AC%E9%A9%97%E3%80%91%E5%89%8D%E5%8F%B0%E5%88%87%E7%89%88-%E6%B8%AC%E9%A9%97)

**Review by Elantris — APPROVED** [review]

**Pull request merged**